src/Resources/contao/dca/tl_calendar_events.php

```diff
@@ -6,21 +6,25 @@
  * @license LGPL-3.0-or-later
  */
 
+use Contao\CoreBundle\Util\PackageUtil;
+
 $dca = &$GLOBALS['TL_DCA']['tl_calendar_events'];
 
 System::getContainer()->get(\HeimrichHannot\EventsBundle\Manager\EventsManager::class)->initCalendarEventsDcaForSubEvents();
 
-System::getContainer()->get(\HeimrichHannot\UtilsBundle\Arrays\ArrayUtil::class)->insertInArrayByName(
-    $dca['list']['operations'],
-    'show', [
-    'feature' => [
-        'label'           => &$GLOBALS['TL_LANG']['tl_calendar_events']['feature'],
-        'icon'            => 'featured.svg',
-        'attributes'      => 'onclick="Backend.getScrollOffset();return AjaxRequest.toggleFeatured(this,%s)"',
-        'button_callback' => [\HeimrichHannot\EventsBundle\EventListener\DataContainer\CalendarEventsListener::class, 'iconFeatured'],
-    ],
-], 0
-);
+if (version_compare(PackageUtil::getContaoVersion(), '4.10', '<')) {
```

Comment:
Could you put this code into a load callback or loadDataContainer hook? Services should not be loaded in a dca file.

Reply:
Sorry, but I also don't have the time at the moment. But there are a lot more service usages in dca files in this extension. I only added the version check for compatibility with 4.x in this PR!

Reply:
I completely unterstand this. Than I will merge it and maybe find some time to refactor later.